otter/cloud_client.py

```diff
@@ -169,10 +212,12 @@ def got_auth((token, catalog)):
         else:
             request_ = add_bind_service(
                 catalog, service_name, region, log, request_)
-        if service_request.json_response:
-            request_ = add_json_response(request_)
         request_ = add_error_handling(
             service_request.success_pred, request_)
+        if service_request.json_response:
+            request_ = add_json_response(request_)
```

Comment:
Yeah. I was also thinking that callers will have to check everytime. That will be a pain. I wonder if providing a `json_response_text_fallback=True` argument would be useful. Maybe not. I am not really convinced either way. So, lets leave it like this :)